### ISSUE TITLE
Memory adapter: `collection` overwrites existing collections

### ISSUE THREAD

**Issue by PandaWhisperer** [pandastrike/pirate#26]
Just noticed this when trying to share a single in-memory database between different source files in the same project.

Let's say I have a file `src/db.coffee`, in which I create a pirate database:

```coffeescript
{Memory} = require "pirate"
module.exports = Memory.Adapter.make()
```

Now I can `require "./db"` in all my other files that need the database, and I'm free to change this to use the Redis adapter later on, without having to update a bunch of files. So far, so good, right?

Now, in `a.coffee`, I want to access the collection "foo":

```coffeescript
db = require "./db"
foo = yield db.collection "foo"
```

Let's say I write some things in there, and in `b.coffee`, I want to read them, so I use the same code to get access to that collection. Right?

Wrong. The `collection` method [creates a new collection *every time* and overwrites any existing ones][1], making it a bad drop-in for a "real" db.

[1]: https://github.com/pandastrike/pirate/blob/es6/src/memory-adapter.coffee#L16-L20

**Comment by PandaWhisperer** [comment]
On a related note, I just noticed that [the Redis adapter's `collection` method does not return a promise][1]. That's not really a big deal when using it with `yield`, but for someone who might be using it from plain old JS it might be confusing to get a promise sometimes, and sometimes not.

This is probably my fault, because I ported the Redis adapter. 

[1]: https://github.com/pandastrike/pirate/blob/es6/src/redis-adapter.coffee#L42-L45

**Comment by dyoder** [comment]
Agreed with both points. We should probably open a separate ticket for the Redis `collection` interface.

**Comment by PandaWhisperer** [comment]
@dyoder you got it: #27. Could you comment there on whether this method should always return a promise?

**Comment by dyoder** [comment]
Commented.

**Comment by PandaWhisperer** [comment]
Closed via #28.
